packages/legend-application-query/src/stores/QueryStudioSDLCState.ts

```diff
@@ -0,0 +1,152 @@
+/**
+ * Copyright (c) 2020-present, Goldman Sachs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { ProjectData } from '@finos/legend-server-depot';
+import { SDLCServerClient } from '@finos/legend-server-sdlc';
+import {
+  type GeneratorFn,
+  assertErrorThrown,
+  guaranteeNonNullable,
+} from '@finos/legend-shared';
+import { action, flow, flowResult, makeObservable, observable } from 'mobx';
+import type { QueryEditorStore } from './QueryEditorStore.js';
+
+export class StudioSDLCInstance {
```

Comment:
Do we _foresee_ this SDLC usage in Query is a one-time thing for the sake of only this flow, or it should become more and more useful. I can't tell, but if it's a one-time thing, I would rather we chuck all this logic into `QueryPromoteToServiceState`.

Also, this flow is not super straightforward, you're refreshing the page to make people go to the authorize URL over on gitlab. Would this disrupt the flow?

I also enjoy the fact that Query app starts with breeze and does not do some OAuth thing, so would this disrupt that? Could we limit the SDLC interaction to only the modal and somehow find a reasonable way to authenticate/authorize for this flow? You could test this by deleting your cache + cookie and try to see the refresh behavior

Reply:
It wouldn't disrupt this flow.. We only call authorize during the `init` flow when the modal is open for the first time or when we are chaning sdlc instances.